tests/console/validate_lvm_with_raid1.pm

```diff
@@ -0,0 +1,89 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple LVM partition validation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "y2_module_consoletest";
+use testapi;
+use utils;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
+
+my $config;
+my @expectedDevs;
+my $expected_num_devs;
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+
+    $config            = get_test_data();
+    @expectedDevs      = (keys %{$config->{disks}});
+    $expected_num_devs = scalar @expectedDevs;
+
+    _lvm_partitioning();
+    _raid1_partitioning();
+    $self->_raid_can_boot_with_deleted_hdd();
+}
+
+sub _lvm_partitioning {
+    my $activelvm = script_output q[lvscan | awk '{print $2}' | sed s/\'//g];
+    assert_equals $config->{lvm}->{lvpath}, $activelvm, "lv name is not the same. Device might not be Active";
+    my $pvname = script_output q[pvscan -s | grep '/dev'];
+    assert_equals $config->{lvm}->{pvname}, $pvname, "pv name is not the same.";
+}
+
+sub _raid1_partitioning {
+    record_info("print info", $config->{raid1}->{name});
+    assert_script_run 'mdadm --detail /dev/' . $config->{raid1}->{name};
+
+    assert_script_run 'grep \'md0 : active ' . $config->{raid1}->{level} . '\' /proc/mdstat';
+
+    my $active_devs = script_output("mdadm --detail /dev/md0 |grep \"Active Devices\" |awk '{ print \$4 }'");
+    assert_equals($active_devs, $expected_num_devs, "Active devices are differ");
+    for (@expectedDevs) {
+        my $line = script_output "mdadm --detail /dev/md0 | awk '{ if((/$_/) && (\$5 ~ /active/) && (\$6 ~ /sync/)) {print}}'";
+        assert_not_null($line, "$_ not active");
+    }
+}
+
+sub _raid_can_boot_with_deleted_hdd {
+    my $self = shift;
+    assert_script_run 'mdadm --manage /dev/md0 --set-faulty /dev/vdd2';
+    $expected_num_devs = '3';
+    my $active_devs = script_output("mdadm --detail /dev/md0 |grep \"Active Devices\" |awk '{ print \$4 }'");
+    assert_equals($expected_num_devs, $active_devs, "Active devices are differ after set faulty device");
+
+    assert_script_run 'mdadm --manage /dev/md0 --remove /dev/vdd2';
+    script_retry "mdadm --detail /dev/md0 | grep 'Active Devices : 3'";
+    my $removedDisk = script_output "mdadm --detail /dev/md0 | awk '{ if((\$4 == 3) && (\$5 ~ /removed/)) {print}}'";
+    assert_not_null($removedDisk, "$removedDisk should be removed");
```

Comment:
`should have been removed` plus `dev`'s and number `3` hardcoded.

Reply:
done